core/src/main/scala/cats/data/XorT.scala

```diff
@@ -0,0 +1,190 @@
+    F.partialFold(value)(axb => axb.fold(_ => Fold.Pass, f))
+
+  def merge[AA >: A](implicit ev: B <:< AA, F: Functor[F]): F[AA] = F.map(value)(_.fold(identity, ev.apply))
+
+  def combine(that: XorT[F, A, B])(implicit F: Apply[F], A: Semigroup[A], B: Semigroup[B]): XorT[F, A, B] =
+    XorT(F.map2(this.value, that.value)(_ combine _))
+
+  def show(implicit show: Show[F[A Xor B]]): String = show.show(value)
+}
+
+object XorT extends XorTInstances with XorTFunctions
+
+trait XorTFunctions {
+
+  final def left[F[_], A, B](fa: F[A])(implicit F: Functor[F]): XorT[F, A, B] = XorT(F.map(fa)(Xor.left))
+
+  final def right[F[_], A, B](fb: F[B])(implicit F: Functor[F]): XorT[F, A, B] = XorT(F.map(fb)(Xor.right))
+
+  final def pure[F[_], A, B](b: B)(implicit F: Applicative[F]): XorT[F, A, B] = right(F.pure(b))
+}
+
+abstract class XorTInstances extends XorTInstances1 {
+
+  /* TODO violates right absorbtion, right distributivity, and left distributivity -- re-enable when MonadCombine laws are split in to weak/strong
+  implicit def xorTMonadCombine[F[_], L](implicit F: Monad[F], L: Monoid[L]): MonadCombine[XorT[F, L, ?]] = {
+    implicit val F0 = F
+    implicit val L0 = L
+    new XorTMonadCombine[F, L] { implicit val F = F0; implicit val L = L0 }
+  }
+  */
+
+  implicit def xorTEq[F[_], L, R](implicit e: Eq[F[L Xor R]]): Eq[XorT[F, L, R]] =
+    // TODO Use Eq.instance on next algebra upgrade
+    new Eq[XorT[F, L, R]] {
+      def eqv(x: XorT[F, L, R], y: XorT[F, L, R]) = e.eqv(x.value, y.value)
+    }
+
+  implicit def xorTShow[F[_], L, R](implicit sh: Show[F[L Xor R]]): Show[XorT[F, L, R]] =
+    functor.Contravariant[Show].contramap(sh)(_.value)
+}
+
+private[data] abstract class XorTInstances1 extends XorTInstances2 {
+  /* TODO violates monadFilter right empty law -- re-enable when MonadFilter laws are split in to weak/strong
+  implicit def xorTMonadFilter[F[_], L](implicit F: Monad[F], L: Monoid[L]): MonadFilter[XorT[F, L, ?]] = {
+    implicit val F0 = F
+    implicit val L0 = L
+    new XorTMonadFilter[F, L] { implicit val F = F0; implicit val L = L0 }
+  }
+  */
+
+ /* TODO delete this when MonadCombine instance is re-enabled */
+ implicit def xorTMonoidK[F[_], L](implicit F: Monad[F], L: Monoid[L]): MonoidK[XorT[F, L, ?]] = {
+    implicit val F0 = F
+    implicit val L0 = L
+    new MonoidK[XorT[F, L, ?]] with XorTSemigroupK[F, L] {
+      implicit val F = F0; implicit val L = L0
+      def empty[A] = XorT.left(F.pure(L.empty))(F)
+    }
+  }
+}
+
+private[data] abstract class XorTInstances2 extends XorTInstances3 {
+  implicit def xorTMonad[F[_], L](implicit F: Monad[F]): Monad[XorT[F, L, ?]] = {
+    implicit val F0 = F
+    new XorTMonad[F, L] { implicit val F = F0 }
+  }
+
+  implicit def xorTSemigroupK[F[_], L](implicit F: Monad[F], L: Semigroup[L]): SemigroupK[XorT[F, L, ?]] = {
+    implicit val F0 = F
+    implicit val L0 = L
+    new XorTSemigroupK[F, L] { implicit val F = F0; implicit val L = L0 }
+  }
```

Comment:
@mpilquist gave his rationale [on Gitter](https://gitter.im/non/cats?at=553b9fe91c7f609d3d25b296).

> Rationale is only that the param names are API in scaladoc

My take on it is that this approach adds a little internal messiness to provide a nicer API, which seems reasonable to me.